packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -49,22 +49,33 @@ void main() {
     );
   }
 
+  Finder findMenuItemButton(String label) {
+    // For each menu items there are two MenuItemButton widgets.
+    // The first one is the real button item in the menu.
+    // The second one is not visible, it is part of _DropdownMenuBody
+    // which is used to compute the dropdown width.
+    return find.widgetWithText(MenuItemButton, label).first;
```

Comment:
We should not use `.first` to get the menu item because we layout the items for the first time to get each items' size but we don't paint them. The `.last` should be the "real" buttons that show in the menu.

Reply:
The order change is a result of the new tree structure.

Before this PR `_DropdownMenuBody` was inside the `MenuAnchor`, with this PR `MenuAnchor` is inside `_DropdownMenuBody`.
Having two occurrences of each widget is somewhat confusing, especially for the testing story, whether the real widget is first or second, I don’t know if there is a way to avoid this.

Do you see any reason (other than not breaking existing tests) why the real item should be the second one and not the first one? In both cases, it seems arbitrary to me?
Nonetheless, If it breaks to many existing tests and it is difficult to update them, we can try to tweak the order, not sure if it is possible.

Before going further, I will investigate if the issue can be fix at `MenuAnchor` level according to https://github.com/flutter/flutter/pull/157496#issuecomment-2449368399